script/vm/visible.lua

```diff
@@ -42,21 +42,33 @@ local function getVisibleType(source)
 
     if type(fieldName) == 'string' then
         local uri = guide.getUri(source)
+        local regengine = config.get(uri, 'Lua.doc.regengine')
+        local function match(patterns)
+            if regengine == "glob" then
+               return glob.glob(patterns)(fieldName)
+            else
+                for i = 1, #patterns do
+                    if string.find(fieldName, patterns[i]) then
+                        return true
+                    end
+                end
+            end
+        end
```

Comment:
I think this function should be declared in the outer scope. Otherwise each call to `getVisibleType()` will create an unnecessary closure function object, and adds burden to garbage collection.

Maybe something like this:
- outer scope
```lua
local function globMatch(patterns, fieldName)
    return glob.glob(patterns)(fieldName)
end

local function luaMatch(patterns, fieldName)
    for i = 1, #patterns do
        if string.find(fieldName, patterns[i]) then
            return true
        end
    end
    return false
end
```
- here
```lua
        local regengine = config.get(uri, 'Lua.doc.regengine')
        local match = regengine  == "glob" and globMatch or luaMatch
        ...
        if #privateNames > 0 and match(privateNames, fieldName) then
```

Reply:
thanks for the advice. :)